pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -649,13 +658,42 @@ func (a *index) inRevision(obj any) bool {
 
 // All return all known workloads. Result is un-ordered
 func (a *index) All() []model.AddressInfo {
-	res := slices.Map(a.workloads.List(), modelWorkloadToAddressInfo)
+	var res []model.AddressInfo
+	// Add all workloads
+	for _, wl := range a.workloads.List() {
+		// If EnableAmbientMultiNetwork is enabled, we only want to add workloads that are local to this cluster
+		// Non cluster local workloads will be added by the service lookup
+		if features.EnableAmbientMultiNetwork && wl.Workload.ClusterId != string(a.ClusterID) {
+			continue
+		}
+		res = append(res, wl.AsAddress)
+	}
+	// Add all services
 	for _, s := range a.services.List() {
+		if features.EnableAmbientMultiNetwork && s.Scope == model.Global {
```

Comment:
What about the case where we don't have a local copy of the service and the only service instance (i.e VIP) is in a remote network AND the service labels don't match a global service scope? Would the hostname represented by that service be local or global, and, more to the point, would it be included in the return values here?

Reply:
If the service labels for a remote service don't match a global scope then the service should be considered locally scoped and should not be returned here

Reply:
Ah you're right; nevermind